compiler/src/dotty/tools/dotc/core/classfile/ClassfileParser.scala

```diff
@@ -798,8 +798,9 @@ class ClassfileParser(
               Array.empty[Byte]
             case Some(jar: ZipArchive) => // We are in a jar
               val jarFile = JarArchive.open(io.File(jar.jpath))
-              try readTastyForClass(jarFile.jpath.resolve(classfile.path))
-              finally jarFile.close()
+              readTastyForClass(jarFile.jpath.resolve(classfile.path))
+              // Do not close the file system as some else might use it later. Once closed it cannot be re-opened.
+              // TODO find a way to safly close the file system or ose some other abstraction
```

Comment:
Why is this an issue now but it was not before?

Reply:
We never reused the same jar file in the same JVM to load some TASTY. With this change, we load the bootstrapped library from the `.tasty` files. During `dotty-bootstrapped/testCompilation` we do reuse this jar all the time.

Reply:
Does this leak memory? If so it definitely needs to be addressed before merging. Also on Windows keeping a file open means that it cannot be deleted, does this apply to this stuff too?

Reply:
It probably leaks but I was not able to see it. I will have another look at this.

Reply:
I changed the implementation and used a classloader to load the tasty files in the jar.